client/src/components/classesPage/ClassCreateView.js

```diff
@@ -80,65 +85,78 @@ function ClassCreateView(props) {
       plain_content: campaignData.plain_content // requires [unsubscribe]
-    }
+    };
 
     // Add new list name
     addList(listData.classnameInput)
       .then(res => {
-        validated.list_ids.push(res.data.id)
-        console.log(`line 83`)
-        console.log(recipientData)
-        return addRecipients(recipientData)
+        validated.list_ids.push(res.data.id);
+        console.log(`line 83`);
+        console.log(recipientData);
+        return addRecipients(recipientData);
       })
 
       // Add new recipients
       .then(res => {
         validated.recipient_ids = [
           ...validated.recipient_ids,
           ...res.data.persisted_recipients
-        ]
-        console.log(`line 94`)
-        console.log(validated.list_ids[0])
-        console.log(validated.recipient_ids)
-        return addContacts(validated.list_ids[0], validated.recipient_ids)
+        ];
+        console.log(`line 94`);
+        console.log(validated.list_ids[0]);
+        console.log('validated recipients:', validated.recipient_ids);
+        return addContacts(validated.list_ids[0], validated.recipient_ids);
       })
 
       // Add all new recipients to list
       .then(res => {
-        validated.selection_code = res.status
+        validated.selection_code = res.status;
         if (validated.selection_code === 201) {
-          console.log(`line 104`)
-          console.log(newRefreshr)
-          return addRefreshr(newRefreshr)
+          console.log(`line 104`);
+          console.log(newRefreshr);
+          return addRefreshr(newRefreshr);
         }
       })
 
       // Send Refreshr object with unix time to be scheduled
       .then(res => {
         if (res.status === 201) {
-          validated.campaign_id = res.data.id
-          console.log(`line 114`)
-          console.log(timeData)
-          console.log(validated.campaign_id)
-          return scheduleRefreshr(timeData, validated.campaign_id)
+          validated.campaign_id = res.data.id;
+          console.log(`line 114`);
+          console.log(timeData);
+          console.log(validated.campaign_id);
+          return scheduleRefreshr(timeData, validated.campaign_id);
         }
       })
 
       // Sucess if all steps complete
       .then(res => {
         if (res.status === 201) {
-          validated.schedule_code = res.status
-          console.log(`Success! Your campaign ${res.data.id} is scheduled for ${res.data.send_at}. Status is "${res.data.status}"!`)
+          validated.schedule_code = res.status;
+          console.log(
+            `Success! Your campaign ${res.data.id} is scheduled for ${
+              res.data.send_at
+            }. Status is "${res.data.status}"!`
+          );
+          setTimeout(() => {
+            submitClassData(
+              listData,
+              validated.list_ids[0],
+              recipientData,
+              campaignData
+            );
+          }, 20000);
```

Comment:
Yea and I think it might take even longer than 20 seconds perhaps.

Sendgrid has a lot of intricacies we need to hammer out.

There was a situation where it says there are 2 recipients on the list but clearly only 1.